Crash when opening app after being paused

```
com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: main
                                        Process: com.amaze.filemanager, PID: 12228
                                        java.lang.RuntimeException: Unable to start activity ComponentInfo{com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.design.widget.NavigationView$SavedState
                                            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2426)
                                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2490)
                                            at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354)
                                            at android.os.Handler.dispatchMessage(Handler.java:102)
                                            at android.os.Looper.loop(Looper.java:148)
                                            at android.app.ActivityThread.main(ActivityThread.java:5443)
                                            at java.lang.reflect.Method.invoke(Native Method)
                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
                                         Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.design.widget.NavigationView$SavedState
                                            at android.os.Parcel.readParcelableCreator(Parcel.java:2432)
                                            at android.os.Parcel.readParcelable(Parcel.java:2358)
                                            at android.view.AbsSavedState.<init>(AbsSavedState.java:57)
                                            at android.view.View$BaseSavedState.<init>(View.java:21467)
                                            at com.amaze.filemanager.ui.views.drawer.CustomNavigationView$SavedState.<init>(CustomNavigationView.java:0)
                                            at com.amaze.filemanager.ui.views.drawer.CustomNavigationView$SavedState.<init>(CustomNavigationView.java:0)
                                            at com.amaze.filemanager.ui.views.drawer.CustomNavigationView$SavedState$1.createFromParcel(CustomNavigationView.java:116)
                                            at com.amaze.filemanager.ui.views.drawer.CustomNavigationView$SavedState$1.createFromParcel(CustomNavigationView.java:114)
                                            at android.os.Parcel.readParcelable(Parcel.java:2367)
                                            at android.os.Parcel.readValue(Parcel.java:2264)
                                            at android.os.Parcel.readSparseArrayInternal(Parcel.java:2675)
                                            at android.os.Parcel.readSparseArray(Parcel.java:1967)
                                            at android.os.Parcel.readValue(Parcel.java:2321)
                                            at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
                                            at android.os.BaseBundle.unparcel(BaseBundle.java:221)
                                            at android.os.Bundle.getSparseParcelableArray(Bundle.java:856)
                                            at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2033)
                                            at android.app.Activity.onRestoreInstanceState(Activity.java:1008)
                                            at android.app.Activity.performRestoreInstanceState(Activity.java:963)
                                            at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1186)
                                            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2399)
                                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2490) 
                                            at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354) 
                                            at android.os.Handler.dispatchMessage(Handler.java:102) 
                                            at android.os.Looper.loop(Looper.java:148) 
                                            at android.app.ActivityThread.main(ActivityThread.java:5443) 
                                            at java.lang.reflect.Method.invoke(Native Method) 
                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728) 
                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618) 
```

--- a/app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java
+++ b/app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java
@@ -1,6 +1,7 @@
 package com.amaze.filemanager.ui.views.drawer;
 
 import android.content.Context;
+import android.os.Build;
 import android.os.Parcel;
 import android.os.Parcelable;
 import android.support.annotation.NonNull;
@@ -65,6 +66,10 @@ public class CustomNavigationView extends NavigationView
 
     @Override
     public Parcelable onSaveInstanceState() {
+        if (!doesNavigationViewSavedStateExist()) {
+            return super.onSaveInstanceState();
+        }
+
         //begin boilerplate code that allows parent classes to save state
         Parcelable superState = super.onSaveInstanceState();
 
@@ -78,6 +83,11 @@ public class CustomNavigationView extends NavigationView
 
     @Override
     public void onRestoreInstanceState(Parcelable state) {
+        if(!doesNavigationViewSavedStateExist()) {
+            super.onRestoreInstanceState(state);
+            return;
+        }
+
         //begin boilerplate code so parent classes can restore state
         if(!(state instanceof SavedState)) {
             super.onRestoreInstanceState(state);
@@ -89,6 +99,20 @@ public class CustomNavigationView extends NavigationView
         //end
 
         this.checkedId = ss.selectedId;
+    }
+
+    /**
+     * This is a hack, when the SavedState class is unmarshalled a "ClassNotFoundException"
+     * will be thrown (the actual class not found is "android.support.design.widget.NavigationView$SavedState")
+     * and I seem to only be able to replicate on Marshmallow.
+     * Trying to find the class and returning false if Class.forName() throws "ClassNotFoundException"
+     * doesn't work because the class seems to have been loaded with the current loader
+     * (not the one the unmarshaller uses); of course I have no idea of what any of this means so
+     * I could be wrong.
+     * For the crash see https://github.com/TeamAmaze/AmazeFileManager/issues/1101.
+     */
+    public boolean doesNavigationViewSavedStateExist() {
+        return Build.VERSION.SDK_INT != Build.VERSION_CODES.M;
     }
 
     static class SavedState extends BaseSavedState {

